core/src/main/java/com/netease/arctic/utils/SchemaUtil.java

```diff
@@ -85,11 +85,23 @@ public static Schema fillUpIdentifierFields(
   public static Schema selectInOrder(Schema baseSchema, List<String> fieldNames) {
     Preconditions.checkNotNull(fieldNames);
     Preconditions.checkNotNull(baseSchema);
+    validateSchemaFields(baseSchema, fieldNames);
 
     int schemaId = baseSchema.schemaId();
     List<Types.NestedField> fields =
         fieldNames.stream().map(baseSchema::findField).collect(Collectors.toList());
 
     return new Schema(schemaId, fields);
   }
+
+  private static void validateSchemaFields(Schema schema, List<String> requiredFields) {
+    List<String> existingFields =
```

Comment:
Maybe we can reuse `Schema#identifierFieldNames` like this:
```
Set<String> existingFields = schema.identifierFieldNames();
```

Reply:
As I have tested, Set<String> existingFields = schema.identifierFieldNames() doesn't work.
![image](https://github.com/NetEase/amoro/assets/5411862/290058e8-076d-4978-bb9a-24a5ef575f74)

Reply:
Sorry, I realize that my previous advice was incorrect, and I apologize for that.

But It seems that `Set` is a better option than `List` here, what do you think? @tcodehuber